faiss/utils/utils.h

```diff
@@ -37,6 +37,9 @@ std::string get_compile_options();
  * Get some stats about the system
  **************************************************/
 
+// Expose FAISS version as a string
+std::string get_version();
```

Comment:
It feels a bit of a waste to produce an allocated string that is already known at compile time. It also does not contribute to exposing this version to the C API.

Reply:
> It also does not contribute to exposing this version to the C API

I am not sure I understand, can you explain?

Reply:
`std::string get_version()` is not a function signature compatible with C due to the C++ string type.

Ultimately, this PR claims to extend the C API while not touching the C API at all. :)